crates/wit-parser/src/lib.rs

```diff
@@ -266,14 +285,24 @@ pub struct IncludeName {
 
 /// The key to the import/export maps of a world. Either a kebab-name or a
 /// unique interface.
-#[derive(Debug, Clone, PartialEq, Eq, Hash)]
+#[derive(Debug, Clone, PartialEq, Eq, Hash, Serialize)]
+#[serde(into = "String")]
 pub enum WorldKey {
     /// A kebab-name.
     Name(String),
     /// An interface which is assigned no kebab-name.
     Interface(InterfaceId),
 }
 
+impl From<WorldKey> for String {
+    fn from(key: WorldKey) -> String {
+        match key {
+            WorldKey::Name(name) => name,
+            WorldKey::Interface(id) => format!("interface-{}", id.index()),
```

Comment:
This may also be what @ydnar is referring to just above this, but I think it'd be good to separate these two ariants a it better than with an `interface-` prefix since that looks a lot like a kebab-name

Reply:
Agreed. Since this is just an internal reference to an `Interface` I think this can be serialized to a JSON number.

Reply:
Upon further investigation, I think we need to have a unique string-serializable key for the `IndexMap<WorldKey, WorldItem>` maps in `World`, given that interfaces may be unnamed:

```rs
    /// Optionally listed name of this interface.
    ///
    /// This is `None` for inline interfaces in worlds.
    pub name: Option<String>,
```

@alexcrichton are you OK with the kebab-case `interface-$N` form or is there another form you’d prefer?